core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -195,6 +195,24 @@ package object config {
         "configured to be at least 10 MiB.")
       .createWithDefaultString("128m")
 
+  private[spark] val EVENT_LOG_ROLLING_MAX_FILES_TO_RETAIN =
+    ConfigBuilder("spark.eventLog.rolling.maxFilesToRetain")
+      // TODO: remove this when integrating compactor with FsHistoryProvider
+      .internal()
+      .doc("The maximum number of event log files which will be retained as non-compacted. " +
+        "By default, all event log files will be retained. Please set the configuration " +
+        s"and ${EVENT_LOG_ROLLING_MAX_FILE_SIZE.key} accordingly if you want to control " +
+        "the overall size of event log files.")
+      .intConf
+      .checkValue(_ > 0, "Max event log files to retain should be higher than 0.")
```

Comment:
Couldn't it be 0, potentially? So you only keep one last compacted log file, instead of one compacted + one normal.

Reply:
If I'm not missing, it can't be, as setting this 0 means we allow compacting the event log file which running app is writing now. It also doesn't make sense for finished app as well, since all jobs should have been finished when app is finished - so compaction would lose all job-related events.

Reply:
Hmm, I see. Merging the last two logs might still result in some savings, but probably not worth it if it means having extra logic to say "don't filter out things that started in the last log file".

On a side note, compaction currently overrides all the "retained blah" configurations; so at the end you may end up, for example, with very few jobs in the Spark UI, whereas the retention settings say you should keep the last 1000 jobs or so.

Reply:
> On a side note, compaction currently overrides all the "retained blah" configurations

Yeah, that's actually the point I have been thinking about, as end users may see only few of jobs if they set this to 1 and the log files are just compacted.

Maybe we could improve event filter to configure max retained jobs on compact file so that some of finished jobs can be still tracked (`live jobs + finished jobs <= max retained jobs`). It cannot be strictly max, as live jobs could have been more than the configured value, so may need to have better name.

Looks like it could be further TODO worth filing JIRA issue.